Neos.Flow/Classes/Reflection/ReflectionService.php

```diff
@@ -1453,6 +1453,12 @@ protected function expandType(ClassReflection $class, $type)
             return $this->expandType($class, $type) . '<' . $this->expandType($class, $elementType) . '>' . ($isNullable ? '|null' : '');
         }
 
+        // expand SomeElementType[]" to "array<\ElementTypeNamespace\SomeElementType>"
+        if (strpos($typeWithoutNull, '[]') === strlen($typeWithoutNull) - 2) {
```

Comment:
```suggestion
        if (substr_compare($typeWithoutNull, '[]', -2, 2) === 0) {
```
Is more understandable and also more performant, since it doesn't need to go through all of $typeWithoutNull to possibly find '[]'. Also, `strrpos` would probably be more correct - even though a type with multiple `[]` is invalid anway.